mlflow/store/file_store.py

```diff
@@ -315,7 +315,7 @@ def create_run(self, experiment_id, user_id, run_name, source_type,
         run_uuid = uuid.uuid4().hex
         artifact_uri = self._get_artifact_dir(experiment_id, run_uuid)
         run_info = RunInfo(run_uuid=run_uuid, experiment_id=experiment_id,
-                           name="",
+                           name="" if run_name is None else run_name,
```

Comment:
This is definitely confusing, but we actually deprecated the run name field -- we'll probably remove it in time for MLflow 1.0. Instead, we use the tag MLFLOW_RUN_NAME tag below. We'll want to update the implementation to read this.